frontend/src/stores/levels/actions.ts

```diff
@@ -0,0 +1,34 @@
+import { levels } from './state'
+import { get } from '@/services/http'
+
+export interface Level {
+  id: number
+  name: string
+  minCoefficient: number
+  maxCoefficient: number
+  link: string
+}
+
+export async function fetchLevelsByRoleId(roleId: string): Promise<Level[]> {
+  const data: Level[] = await get(`/levels?role_id=${roleId}`)
+
+  const levelsData: Level[] = data.map((level) => ({
+    id: level.id,
+    name: level.name,
+    minCoefficient: 0,
+    maxCoefficient: 0,
+    link: ''
+  }))
+
+  levels.value = levelsData
```

Comment:
You don't need a state for levels. You can put the levels of a role inside the role state. And then, when you call this method, you should verify if the levels of the role already exists (in case yes, you just return)